### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by citec-47**
Hello 😁👋, In this PR, we did the unit testing for the classes we created with the following requirements 📑

General requirements
✅ Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
✅ Make sure that you used correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
✅ Make sure that you documented your work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).

Ruby requirements
✅ Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

Project requirements
Add unit tests for all the methods in all the classes:
✅ Book
✅ Person
✅ Student
✅ Teacher
✅ Classroom
✅ Rental
✅ all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  it 'correct age method' do
+    expect(@person.send('of_age?')).to be true
+  end
+
+  it 'can use services' do
+    expect(@person.can_use_services?).to be true
+  end
+end
```
Comment (cynthiainga) [inline spec/person_spec.rb]: - [x] You should test the scenario of the constructor method sets a default value for the optional name parameter if it is not provided 🙃 
- [x] Also the scenario of the `can_use_services?` method returns false if under age and no parent permission 😁 
- [x] You should test also the scenario of the `can_use_services?` method returns true if has parent permission 😁

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @citec-47 & @zunairkhan811, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] All implemented tests are passing ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/person_spec.rb (hunk 1 above)
Comment (zunairkhan811) [reply spec/person_spec.rb]: Thanks! @cynthiainga, more tests have been added.

**Review by mckent05 — APPROVED** [review]
Hi @citec-47, 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏 
- Good job adding tests to your classes 👍 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**